src/components/checkbox/checkbox.scss

```diff
@@ -0,0 +1,401 @@
+  position: relative;
+  vertical-align: middle;
+  white-space: nowrap;
+  width: $md-checkbox-size;
+
+  [dir="rtl"] & {
+    margin:{
+      left: $md-checkbox-item-spacing;
+      right: auto;
+    }
+  }
+}
+
+.md-checkbox-frame {
+  @extend %md-checkbox-background;
+
+  background-color: transparent;
+  border: $md-checkbox-border-width solid $md-checkbox-border-color;
+  transition: border-color $md-checkbox-transition-duration $linear-out-slow-in-timing-function;
+  will-change: border-color;
+}
+
+.md-checkbox-background {
+  @extend %md-checkbox-background;
+
+  align-items: center;
+  background-color: $md-checkbox-background-color;
+  opacity: 0;
+  display: inline-flex;
+  justify-content: center;
+  transition: background-color $md-checkbox-transition-duration $linear-out-slow-in-timing-function,
+              opacity $md-checkbox-transition-duration $linear-out-slow-in-timing-function;
+  will-change: background-color, opacity;
+}
+
+.md-checkbox-checkmark {
+  @extend %md-checkbox-cover-element;
+  @extend %md-checkbox-mark;
+
+  fill: $md-checkbox-mark-color;
+  width: 100%;
+}
+
+.md-checkbox-checkmark-path {
+  // !important is needed here because a stroke must be set as an attribute on the SVG in order
+  // for line animation to work properly.
+  stroke: $md-checkbox-mark-color !important;
+  stroke: {
+    dashoffset: $_md-checkbox-mark-path-length;
+    dasharray: $_md-checkbox-mark-path-length;
+    width: $md-checkbox-mark-stroke-size;
+  }
+}
+
+.md-checkbox-mixedmark {
+  @extend %md-checkbox-mark;
+
+  background-color: $md-checkbox-mark-color;
+  height: $md-checkbox-mark-stroke-size;
+  opacity: 0;
+  transform: scaleX(0) rotate(0deg);
+}
+
+.md-checkbox-align-end {
+  .md-checkbox-inner-container {
+    order: 1;
+    margin: {
+      left: $md-checkbox-item-spacing;
+      right: auto;
+    }
+
+    [dir="rtl"] & {
+      margin: {
+        left: auto;
+        right: $md-checkbox-item-spacing;
+      }
+    }
+  }
+}
+
+.md-checkbox-checked {
+  .md-checkbox-checkmark {
+    opacity: 1;
+  }
+
+  .md-checkbox-checkmark-path {
+    stroke-dashoffset: 0;
+  }
+
+  .md-checkbox-mixedmark {
+    transform: scaleX(1) rotate(-45deg);
+  }
+
+  .md-checkbox-background {
+    opacity: 1;
+  }
+}
+
+.md-checkbox-indeterminate {
+  .md-checkbox-background {
+    opacity: 1;
+  }
+
+  .md-checkbox-checkmark {
+    opacity: 0;
+    transform: rotate(45deg);
+  }
+
+  .md-checkbox-checkmark-path {
+    stroke-dashoffset: 0;
+  }
+
+  .md-checkbox-mixedmark {
+    opacity: 1;
+    transform: scaleX(1) rotate(0deg);
+  }
+}
+
+.md-checkbox-disabled {
+  // NOTE(traviskaufman): While the spec calls for translucent blacks/whites for disabled colors,
+  // this does not work well with elements layered on top of one another. To get around this we
+  // blend the colors together based on the base color and the theme background.
+  $white-30pct-opacity-on-dark-theme: #686868;
+  $black-26pct-opacity-on-light-theme: #b0b0b0;
```

Comment:
I'm not sure that would work since it looks like it produces a non-opaque color, however it's a really cool function :smile: (I starred it!). Since the disabled colors are theme-independent, and the variable names seem semantic enough, I don't think there's a major need to include it.